src/AuthLDAP.php

```diff
@@ -4370,4 +4382,26 @@ final public static function buildError($ds, string $message): string
         );
         return $message;
     }
+
+    private function removeDefaultFromOtherItems(int $authldaps_id): void
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        if (isset($this->fields['is_default']) && (int) $this->fields["is_default"] === 1) {
+            // remove from authmails tables
+            $DB->update(
```

Comment:
I know you copied the existing code, but we should avoid using low level methods directly to update an item, as they prevent plugins hooks to be able to handle the corresponding operations. You should replace `DBmysql::update()` by `CommonDBTM::update()` here. It requires some refactoring, as you will have first to fetch the existing default server, if any, and the update it.

Same remark for the `AuthMail` update.

Reply:
:/ It doesn't look like CommonDBTM::update() give the ability to update multiple items with a `where` criteria.